lib/src/Uncertainty/Algorithm/MetaModel/openturns/MetaModelAlgorithm.hxx

```diff
@@ -90,7 +90,7 @@ public:
   void load(Advocate & adv) override;
 
   /** Recover the distribution */
-  static Distribution BuildDistribution(const Sample & inputSample);
+  static ComposedDistribution BuildDistribution(const Sample & inputSample);
```

Comment:
why is this needed ?

Reply:
The `ComposedDistribution` has a pretty-print, after #2282.
But to express it into the examples, the type of the object has to correspond to the class which has a pretty-print feature. After the modification in the current PR, the distribution truly is a `ComposedDistribution`, which produces:

![image](https://github.com/openturns/openturns/assets/31351465/32f9eaec-a5da-4a76-951e-52081a13deaf)

Reply:
in that case the pretty-print method must be added to the Distribution interface class, ant that change must be reverted

Reply:
I added a pretty-print to the Distribution at d7f311dc969d301baf5869c638d7d26dc80dbf9e. Notice, however, that this cannot be a replacement for this particular code. This is because the information contained in the ComposedDistribution is much more specific than the information in a Distribution: a ComposedDistribution has a list of 1D marginals, while a Distribution has not. The information contained in the attributes of the Distribution are the dimension, the description, the range and the copula boolean. The pretty-print is an itemized list, but not more.

Reply:
no, the pretty-print will be delegated to ComposedDistribution by inheritance and is done in the implementation so that change is not needed

Reply:
I was not sure to get the point, so I tested several combinations. There are two options:
- Option 1. The `MetaModelAlgorithm.BuildDistribution()` returns a `ComposedDistribution`.
- Option 2. The `MetaModelAlgorithm.BuildDistribution()` returns a `Distribution`.

There are two sub-options:
- Option A. Implement `Distribution.__Distribution_repr_html` in `Distribution.i`.
- Option B. Do not implement `Distribution.__Distribution_repr_html` in `Distribution.i`.

In all cases, I leave the `_repr_html_` implemented for `ComposedDistribution.i`.

I consider the Python script:
```python
import openturns as ot
inputSample = ot.Normal(2).getSample(1000)
distribution = ot.MetaModelAlgorithm.BuildDistribution(inputSample)
print("distribution = ", distribution)
print("type(distribution) = ", type(distribution))
print("distribution (HTML)")
print(distribution._repr_html_())
print("distribution (Markdown)")
print(distribution._repr_markdown_())
```

- Option 1.A. The `MetaModelAlgorithm.BuildDistribution()` returns a `ComposedDistribution` and I implement `Distribution.__Distribution_repr_html` in `Distribution.i`.
	Script output:
```bash
$ python3 metamodel_builddistribution.py 
distribution =  ComposedDistribution(Normal(mu = -0.0268588, sigma = 0.985838), Normal(mu = 0.00246667, sigma = 1.00015), IndependentCopula(dimension = 2))
type(distribution) =  <class 'openturns.model_copula.ComposedDistribution'>
distribution (HTML)
<ul>
  <li>dimension: 2</li>
  <li>copula: IndependentCopula</li>
</ul>

<table>
  <tr>
    <th>Index</th>
    <th>Variable</th>
    <th>Distribution</th>
  </tr>
  <tr>
    <td>0</td>
    <td>X0</td>
    <td>Normal(mu = -0.0268588, sigma = 0.985838)</td>
  </tr>
  <tr>
    <td>1</td>
    <td>X1</td>
    <td>Normal(mu = 0.00246667, sigma = 1.00015)</td>
  </tr>
</table>

distribution (Markdown)
ComposedDistribution
- name=ComposedDistribution
- dimension=2
- description=[X0,X1]
- copula=IndependentCopula(dimension = 2)

| Index | Variable | Distribution                              |
|-------|----------|-------------------------------------------|
|     0 |       X0 | Normal(mu = -0.0268588, sigma = 0.985838) |
|     1 |       X1 | Normal(mu = 0.00246667, sigma = 1.00015)  |
```
Conclusion: this is the expected behavior.

- Option 1.B. The `MetaModelAlgorithm.BuildDistribution()` returns a `ComposedDistribution` and I do not implement `Distribution.__Distribution_repr_html` in `Distribution.i`.
	Script output:
```bash
$ python3 metamodel_builddistribution.py 
distribution =  ComposedDistribution(Normal(mu = -0.0268588, sigma = 0.985838), Normal(mu = 0.00246667, sigma = 1.00015), IndependentCopula(dimension = 2))

type(distribution) =  <class 'openturns.model_copula.ComposedDistribution'>

distribution (HTML)
<ul>
  <li>dimension: 2</li>
  <li>copula: IndependentCopula</li>
</ul>

<table>
  <tr>
    <th>Index</th>
    <th>Variable</th>
    <th>Distribution</th>
  </tr>
  <tr>
    <td>0</td>
    <td>X0</td>
    <td>Normal(mu = -0.0268588, sigma = 0.985838)</td>
  </tr>
  <tr>
    <td>1</td>
    <td>X1</td>
    <td>Normal(mu = 0.00246667, sigma = 1.00015)</td>
  </tr>
</table>

distribution (Markdown)
ComposedDistribution
- name=ComposedDistribution
- dimension=2
- description=[X0,X1]
- copula=IndependentCopula(dimension = 2)

| Index | Variable | Distribution                              |
|-------|----------|-------------------------------------------|
|     0 |       X0 | Normal(mu = -0.0268588, sigma = 0.985838) |
|     1 |       X1 | Normal(mu = 0.00246667, sigma = 1.00015)  |
```
Conclusion: this is the expected behavior, as before. But `Distribution` does not have a pretty-print anymore. This is not a problem, because this pretty-print is not very interesting.

- Option 2.A.The `MetaModelAlgorithm.BuildDistribution()` returns a `Distribution` and I implement `Distribution.__Distribution_repr_html in Distribution.i`
	Script output:
```bash
$ python3 metamodel_builddistribution.py 
distribution =  ComposedDistribution(Normal(mu = -0.0268588, sigma = 0.985838), Normal(mu = 0.00246667, sigma = 1.00015), IndependentCopula(dimension = 2))

type(distribution) =  <class 'openturns.model_copula.Distribution'>

distribution (HTML)
<ul>
  <li>dimension: 2</li>
  <li>description: [X0,X1]</li>
  <li>range: ]-inf (-7.56914), (7.51542) +inf[
]-inf (-7.64933), (7.65426) +inf[</li>
  <li>isCopula: False</li>
</ul>

distribution (Markdown)
ComposedDistribution
- name=ComposedDistribution
- dimension=2
- weight=1
- range=]-inf (-7.56914), (7.51542) +inf[
]-inf (-7.64933), (7.65426) +inf[
- description=[X0,X1]
- isParallel=true
- isCopula=false
```
Conclusion. The Markdown method is delegated from `Distribution` to `ComposedDistribution`. The HTML method from `Distribution.i` takes the lead and hides the method from `ComposedDistribution.i`. 
The HTML output is disappointing in this case, because the ComposedDistribution is not pretty-printed anymore.

- Option 2.B. The `MetaModelAlgorithm.BuildDistribution()` returns a `Distribution` and I do not implement `Distribution.__Distribution_repr_html` in `Distribution.i`
Script output:
```bash
$ python3 metamodel_builddistribution.py 
distribution =  ComposedDistribution(Normal(mu = -0.0268588, sigma = 0.985838), Normal(mu = 0.00246667, sigma = 1.00015), IndependentCopula(dimension = 2))

type(distribution) =  <class 'openturns.model_copula.Distribution'>

distribution (HTML)
<p>ComposedDistribution(Normal(mu = -0.0268588, sigma = 0.985838), Normal(mu = 0.00246667, sigma = 1.00015), IndependentCopula(dimension = 2))</p>

distribution (Markdown)
Traceback (most recent call last):
  File "/home/devel/Documents/metamodel_builddistribution.py", line 9, in <module>
    print(distribution._repr_markdown_())
AttributeError: 'Distribution' object has no attribute '_repr_markdown_'
```
Conclusion. The HTML output is disappointing in this case, because the `ComposedDistribution` is not pretty-printed anymore. There is no Markdown output.

I think that the behavior that you want can be obtained only if the HTML pretty-print is produced right from the C++ which leave some space for the the inheritance to take place. This is not the case in the current implementation, where the HTML is produced from the Python layer for these objects.

Reply:
yes, it should be done in the c++ side, sorry this wasnt clear enough

Reply:
This part of the PR is now moved to #2447. This is why I resolve this conversation.

Reply:
Oups, the code is still in the PR.